lib/twilioconnection.js

```diff
@@ -284,8 +296,10 @@ class TwilioConnection extends StateMachine {
     if (this.state !== 'connecting') {
       return;
     }
-    this._log.warn(`Closing: ${WS_CLOSE_WELCOME_TIMEOUT} - "welcome" message timeout expired`);
-    this._ws.close(WS_CLOSE_WELCOME_TIMEOUT, '"welcome" message timeout expired');
+    const reason = '"welcome" message timeout expired';
+    this._log.warn(`Closing: ${WS_CLOSE_WELCOME_TIMEOUT} - ${reason}`);
+    this._ws.close(WS_CLOSE_WELCOME_TIMEOUT, reason);
+    this._close({ code: WS_CLOSE_HEARTBEATS_MISSED, reason });
```

Comment:
It looks to me like we always do

```js
this._ws.close(...);
this._close(...);
```

so can we move the `this._ws.close` call into `this._close`, possibly gated on `readyState`?